src/Compilers/CSharp/Portable/Symbols/PublicModel/MethodSymbol.cs

```diff
@@ -0,0 +1,322 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Threading;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols.PublicModel
+{
+    internal sealed class MethodSymbol : Symbol, IMethodSymbol
+    {
+        private readonly Symbols.MethodSymbol _underlying;
+        private ITypeSymbol _lazyReturnType;
+        private ImmutableArray<ITypeSymbol> _lazyTypeArguments;
+        private ITypeSymbol _lazyReceiverType;
+
+        public MethodSymbol(Symbols.MethodSymbol underlying)
+        {
+            Debug.Assert(underlying is object);
+            _underlying = underlying;
+        }
+
+        internal override CSharp.Symbol UnderlyingSymbol => _underlying;
+        internal Symbols.MethodSymbol UnderlyingMethodSymbol => _underlying;
+
+        MethodKind IMethodSymbol.MethodKind
+        {
+            get
+            {
+                switch (_underlying.MethodKind)
+                {
+                    case MethodKind.AnonymousFunction:
+                        return MethodKind.AnonymousFunction;
+                    case MethodKind.Constructor:
+                        return MethodKind.Constructor;
+                    case MethodKind.Conversion:
+                        return MethodKind.Conversion;
+                    case MethodKind.DelegateInvoke:
+                        return MethodKind.DelegateInvoke;
+                    case MethodKind.Destructor:
+                        return MethodKind.Destructor;
+                    case MethodKind.EventAdd:
+                        return MethodKind.EventAdd;
+                    case MethodKind.EventRemove:
+                        return MethodKind.EventRemove;
+                    case MethodKind.ExplicitInterfaceImplementation:
+                        return MethodKind.ExplicitInterfaceImplementation;
+                    case MethodKind.UserDefinedOperator:
+                        return MethodKind.UserDefinedOperator;
+                    case MethodKind.BuiltinOperator:
+                        return MethodKind.BuiltinOperator;
+                    case MethodKind.Ordinary:
+                        return MethodKind.Ordinary;
+                    case MethodKind.PropertyGet:
+                        return MethodKind.PropertyGet;
+                    case MethodKind.PropertySet:
+                        return MethodKind.PropertySet;
+                    case MethodKind.ReducedExtension:
+                        return MethodKind.ReducedExtension;
+                    case MethodKind.StaticConstructor:
+                        return MethodKind.StaticConstructor;
+                    case MethodKind.LocalFunction:
+                        return MethodKind.LocalFunction;
+                    default:
```

Comment:
It would be easier to verify the correctness of this if it were written

``` c#
                switch (_underlying.MethodKind)
                {
                    case MethodKind.AnonymousFunction:
                    case MethodKind.Constructor:
                    case MethodKind.Conversion:
                    case MethodKind.DelegateInvoke:
                    case MethodKind.Destructor:
                    case MethodKind.EventAdd:
                    case MethodKind.EventRemove:
                    case MethodKind.ExplicitInterfaceImplementation:
                    case MethodKind.UserDefinedOperator:
                    case MethodKind.BuiltinOperator:
                    case MethodKind.Ordinary:
                    case MethodKind.PropertyGet:
                    case MethodKind.PropertySet:
                    case MethodKind.ReducedExtension:
                    case MethodKind.StaticConstructor:
                    case MethodKind.LocalFunction:
                        return _underlying.MethodKind;
                    default:
                        throw ExceptionUtilities.UnexpectedValue(_underlying.MethodKind);
                }
```

or even just
``` c#
                return _underlying.MethodKind;
```

Reply:
> It would be easier to verify the correctness of this if it were written

This implementation was moved from Microsoft.CodeAnalysis.CSharp.Symbols.MethodSymbol. I prefer to not make any implementation changes that are not required to reach the stated goal of this PR.

---
In reply to: [339824774](https://github.com/dotnet/roslyn/pull/39498#discussion_r339824774) [](ancestors = 339824774)